packages/story-editor/src/app/story/useStoryReducer/useStoryReducer.js

```diff
@@ -82,10 +82,27 @@ function useStoryReducer(partial) {
   }, [dispatch]);
 
   return {
-    state,
+    state: deepFreeze(state),
     internal,
     api,
   };
 }
 
+function deepFreeze(o) {
+  Object.freeze(o);
+
+  Object.getOwnPropertyNames(o).forEach(function (prop) {
+    if (
+      Object.prototype.hasOwnProperty.call(o, prop) &&
+      o[prop] !== null &&
+      (typeof o[prop] === 'object' || typeof o[prop] === 'function') &&
+      !Object.isFrozen(o[prop])
+    ) {
+      deepFreeze(o[prop]);
+    }
+  });
+
+  return o;
+}
```

Comment:
This is copied in (and slightly modified due to lint rules) from [the `deepFreeze` library](https://github.com/substack/deep-freeze/blob/master/index.js). It seemed like such a small feature that it didn't make a lot of sense to add it as a dependency, but I'm okay with doing that if that's consensus. If we don't do that, this function should be moved to an util somewhere and have some tests added.